### PR TITLE
Anonymize after Authentication

### PR DESCRIPTION
**Pull request by jdetle**
## Changed
- App record now has publishAsUser boolean
- Separated setGithubToken and setPublishAsUser
- DispatchPublishUserGist has clearer logic
- DispatchPublishAnonGist has clearer logic
- Add SET_USER_GITHUB, SET_ANON_GITHUB actions
- Add action/reducer menu tests 
- Add menu spec tests
- Add publish user to dummy store

### CODE REVIEW

**Review by captainsafia — CHANGES_REQUESTED** [review]
Tests need updates.

**Comment by jdetle** [comment]
:shipit:

File: src/main/menu.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ export function authAndPublish(item, focusedWindow) {
   const win = new BrowserWindow({ show: false,
                                   webPreferences: { zoomFactor: 0.75 } });
   if (process.env.AUTHENTICATED) {
-    send(focusedWindow, 'menu:publish:gist');
+    send(focusedWindow, 'menu:github:auth');
```
Comment (rgbkrk) [inline src/main/menu.js:26]: Relying on an environment variable for authenticated is a bit strange here, I need to look into the intent soon.

**Review by rgbkrk — COMMENTED** [review]
I think you can simplify the publish anonymous logic by not making it part of the state tree -- make a new Github() object every time you publish anon, don't override the old one.

**Pull request merged**